L89-core-remove-grpc-insecure-channel-creation-api.md

```diff
@@ -0,0 +1,142 @@
+Remove grpc_insecure_channel_create and grpc_server_add_insecure_http2_port from Core Surface API
+----
+* Author(s): yihuazhang
+* Approver: markdroth
+* Status: {In Review}
+* Implemented in: https://github.com/grpc/grpc/pull/25586
+* Last updated: 2021-11-01
+* Discussion at: https://groups.google.com/g/grpc-io/c/9B4-AHCx_ZM 
+
+## Abstract
+
+* Remove `grpc_insecure_channel_create` and `grpc_server_add_insecure_http2_port` from Core Surface API.
+* Rename `grpc_server_add_secure_http2_port` to `grpc_server_add_http2_port`.
+* Rename `grpc_secure_channel_create` to `grpc_channel_create` and re-order its parameters.
+* Move `grpc_channel_credentials` and `grpc_server_credentials` related APIs from `grpc_security.h` to `grpc.h`.
+* Add `grpc_channel_create_from_fd` and `grpc_server_add_channel_from_fd` to Core Surface API.
+
```

Comment:
Listing the API changes you're planning is definitely a must, but since a big portion of the design discussions were also about the impact on the build targets, it probably makes sense to add a few sentences that describe the impact on existing build targets (e.g. `grpc_unsecure` and `grpc` which play the most important role here).
- let's mention the planned impact on the existing public bazel targets
- let's also mention the impact on the non-bazel targets (which are mostly represented by our cmake targets `grpc` and `grpc_unsecure`.

From you implementation PR, it looks like the `//:grpc` and `//:grpc_unsecure` targets will be kept and `//:grpc_unsecure` won't have a dependency on boringssl.
The cmake targets: it looks like we will still have `grpc` (with unconditional dependency on boringssl/openssl) and `grpc_unsecure` (NO  dependency on boringssl/openssl, also unconditionally).

If the above is correct, the proposal LGTM to me after adding the summary to the abstract.

Reply:
> From you implementation PR, it looks like the //:grpc and //:grpc_unsecure targets will be kept and //:grpc_unsecure won't have a dependency on boringssl.
The cmake targets: it looks like we will still have grpc (with unconditional dependency on boringssl/openssl) and grpc_unsecure (NO dependency on boringssl/openssl, also unconditionally).

That's correct, Jan. I also updated the text to make it more clear. Thanks for pointing it out.